### PR TITLE
terraform-providers: disable CGO, set ldflags

### PR DESCRIPTION
**Pull request by zowoq**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by zowoq** [comment]
@ofborg build terraform-full

**Comment by zowoq** [comment]
@ofborg build terraform-full

File: pkgs/applications/networking/cluster/terraform-providers/default.nix [hunk 1]
```diff
@@ -14,6 +14,9 @@ let
       version = data.version;
       subPackages = [ "." ];
       doCheck = false;
+      # https://github.com/hashicorp/terraform-provider-scaffolding/blob/main/.goreleaser.yml
+      CGO_ENABLED = 0;
+      ldflags = [ "-s" "-w" "-X main.version=${data.version}" "-X main.commit=${data.rev}" ];
```
Comment (zowoq) [inline pkgs/applications/networking/cluster/terraform-providers/default.nix:19]: Even if some upstreams don't set these `ldflags` I don't see them causing any problems.

File: pkgs/applications/networking/cluster/terraform-providers/default.nix [hunk 2]
```diff
@@ -14,6 +14,9 @@ let
       version = data.version;
       subPackages = [ "." ];
       doCheck = false;
+      # https://github.com/hashicorp/terraform-provider-scaffolding/blob/main/.goreleaser.yml
```
Comment (nh2) [inline pkgs/applications/networking/cluster/terraform-providers/default.nix:17]: Pin URL (press <kbd>Y</kbd> on that Github page to get a stable link), write explicit motivation in addition to the link, like:

```suggestion
      # Upstream says:
      # > goreleaser does not work with CGO
      
      # from: https://github.com/hashicorp/terraform-provider-scaffolding/blob/a8ac8375a7082befe55b71c8cbb048493dd220c2/.goreleaser.yml
```

**Comment by zimbatm** [comment]
I'm getting a weird error with nixpkgs-review:
```
$ nix-shell /home/zimbatm/.cache/nixpkgs-review/pr-153345-3/shell.nix
this derivation will be built:
  /nix/store/9v9vz5xmqxvzyslvckb6w0nphcvjmahg-env.drv
building '/nix/store/9v9vz5xmqxvzyslvckb6w0nphcvjmahg-env.drv'...
error: The store path /nix/store/zn1pyfz2ia166qbz558x3ak8jarhjs8l-terraform-plugin-test is a file and can't be merged into an environment using pkgs.buildEnv! at /nix/store/lf2djfrfli98wn204v5dj6r04hj901f3-builder.pl line 122.
error: builder for '/nix/store/9v9vz5xmqxvzyslvckb6w0nphcvjmahg-env.drv' failed with exit code 2;
       last 1 log lines:
       > error: The store path /nix/store/zn1pyfz2ia166qbz558x3ak8jarhjs8l-terraform-plugin-test is a file and can't be merged into an environment using pkgs.buildEnv! at /nix/store/lf2djfrfli98wn204v5dj6r04hj901f3-builder.pl line 122.
       For full logs, run 'nix log /nix/store/9v9vz5xmqxvzyslvckb6w0nphcvjmahg-env.drv'.
```

**Comment by zowoq** [comment]
I can't reproduce an error with `terraform_plugins_test`.

@ofborg build terraform_plugins_test

**Pull request merged**